src/cart/crt_ctl.c

```diff
@@ -75,6 +75,135 @@ static int verify_ctl_in_args(struct crt_ctl_ep_ls_in *in_args)
 	return rc;
 }
 
+int
+crt_ctl_fill_buffer_cb(d_list_t *rlink, void *arg)
+{
+	struct crt_uri_item	*ui;
+	int			*idx;
+	char			*buf;
+	struct crt_uri_cache	*uri_cache = arg;
+	int			 i;
+	int			 uri_size;
+	int			 rc = 0;
+
+	D_ASSERT(rlink != NULL);
+	D_ASSERT(arg != NULL);
+
+	ui = crt_ui_link2ptr(rlink);
+
+	D_MUTEX_LOCK(&ui->ui_mutex);
+	for (i = 0; i < CRT_SRV_CONTEXT_NUM; i++) {
+		if (ui->ui_uri[i] != NULL) {
+			char *tmp_uri;
+
+			tmp_uri = ui->ui_uri[i];
+
+			idx = &uri_cache->info_idx;
+
+			buf = uri_cache->info_buf;
+
+			/* Pack rank */
+			*((d_rank_t *)(buf + *idx)) = ui->ui_rank;
+			*idx += sizeof(d_rank_t);
+
+			/* Pack tag */
+			*((uint32_t *)(buf + *idx)) = i;
+			*idx += sizeof(uint32_t);
+
+			/* Pack uri size */
+			uri_size = strlen(tmp_uri) + 1;
+			*((uint32_t *)(buf + *idx)) = uri_size;
```

Comment:
What happens if tmp_uri is not a multiple of 4?  Would it mean that for future loops of this code the integer writes were non-aligned?